mbuild/compound.py

```diff
@@ -2235,6 +2346,208 @@ def _energy_minimize_openbabel(
                                 particle, particle.name
                             )
                         )
+        # initialize constraints
+        ob_constraints = openbabel.OBFFConstraints()
+
+        if distance_constraints is not None:
+            # check the dimensions of the array
+            # so that a user passed a 1D array corresponding to only a the
+            # tuple for the pair and the distance, it doesn't try to
+            # loop over the list
+
+            temp1 = np.array(distance_constraints)
+            if len(temp1.shape) == 1:
+                distance_constraints = [distance_constraints]
+
+            for con_temp in distance_constraints:
+                p1 = con_temp[0][0]
+                p2 = con_temp[0][1]
+
+                if id(p1) == id(p2):
+                    raise MBuildError(
+                        f"Cannot create a constraint between a particle and itself: {p1} {p2} ."
+                    )
+                if len(p1.children) != 0 or len(p2.children) != 0:
+                    raise MBuildError(
+                        f"Constraints must correspond to individual atoms in Compounds."
+                    )
+
+                p1_in_compound = False
+                p2_in_compound = False
+                for successor in self.successors():
+                    if id(p1) == id(successor):
+                        p1_in_compound = True
+                    if id(p2) == id(successor):
+                        p2_in_compound = True
+                if p1_in_compound == False and p2_in_compound == True:
```

Comment:
Yeah that is cleaner...I structured it initially as I did to fully evaluate the inputs as a whole before throwing an error (as I get annoyed with partial checks)...although even still, it will potentially error out before evaluating all the inputs if multiple distance constraints are given (not sure it is worth the effort to make a validator/aggregation function as this isn't like a major feature). Cleaner more compact code is probably the way to go.